src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -15,18 +16,25 @@
 
 (trapperkeeper/defservice jruby-puppet-pooled-service
                           jruby/JRubyPuppetService
-                          [[:ConfigService get-config]
-                           [:ShutdownService shutdown-on-error]
-                           [:PuppetProfilerService get-profiler]
-                           [:PoolManagerService create-pool]]
+                          {:required [[:ConfigService get-config]
+                                      [:ShutdownService shutdown-on-error]
+                                      [:PuppetProfilerService get-profiler]
+                                      [:PoolManagerService create-pool]]
+                           :optional [MetricsService]}
```

Comment:
Putting the setting under `http-client` and defaulting it to `true` makes sense to me.